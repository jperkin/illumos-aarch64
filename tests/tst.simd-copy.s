/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.6.5 AdvSIMD copy
 * C3.6.7 AdvSIMD scalar copy
 */
.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/* DUP <Vd>.<T>, <Vn>.<Ts>[<index>]*/
	dup	v0.8b, v1.b[0]
	dup	v0.16b, v1.b[15]
	dup	v0.4h, v1.h[0]
	dup	v0.8h, v1.h[7]
	dup	v0.2s, v1.s[0]
	dup	v0.4s, v1.s[3]
	dup	v0.2d, v1.d[0]
	dup	v0.2d, v1.d[1]
	/* DUP <Vd>.<T>, <R><n> */
	dup	v0.8b, w0
	dup	v31.16b, wzr
	dup	v0.4h, w0
	dup	v31.8h, wzr
	dup	v0.2s, w0
	dup	v31.4s, wzr
	dup	v0.2d, x0
	dup	v31.2d, xzr
	/* SMOV <Rd>, <Vn>.<Ts>[<index>] */
	/* UMOV <Rd>, <Vn>.<Ts>[<index>] */
	smov	w0, v0.b[0]
	smov	wzr, v31.h[7]
	smov	x0, v0.b[15]
	smov	xzr, v31.h[7]
	smov	xzr, v31.s[3]
	umov	w0, v0.b[0]
	umov	wzr, v31.h[7]
	umov	x0, v0.d[0]
	umov	xzr, v31.d[1]
	/* INS <Vd>.<Ts>[<index>], <R><n> */
	ins	v0.b[15], w0
	ins	v31.h[7], wzr
	ins	v31.s[3], w30
	ins	v31.d[1], xzr
	/* INS <Vd>.<Ts>[<index1>], <Vn>.<Ts>[<index2>] */
	ins	v0.b[15], v31.b[0]
	ins	v0.h[7], v31.h[0]
	ins	v0.s[3], v31.s[0]
	ins	v0.d[1], v31.d[0]
	/* DUP <V><d>, <Vn>.<T>[<index>] */
	dup	b0, v0.8b[0]
	dup	b31, v31.16b[15]
	dup	h0, v0.4h[0]
	dup	h31, v31.8h[7]
	dup	s0, v0.2s[0]
	dup	s31, v31.4s[3]
	dup	d0, v0.1d[0]
	dup	d31, v31.2d[1]
.size libdis_test, [.-libdis_test]
