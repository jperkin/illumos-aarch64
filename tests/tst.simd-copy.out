    libdis_test:      20 04 01 0e  dup v0.8b, v1.b[0]
    libdis_test+0x4:  20 04 1f 4e  dup v0.16b, v1.b[15]
    libdis_test+0x8:  20 04 02 0e  dup v0.4h, v1.h[0]
    libdis_test+0xc:  20 04 1e 4e  dup v0.8h, v1.h[7]
    libdis_test+0x10: 20 04 04 0e  dup v0.2s, v1.s[0]
    libdis_test+0x14: 20 04 1c 4e  dup v0.4s, v1.s[3]
    libdis_test+0x18: 20 04 08 4e  dup v0.2d, v1.d[0]
    libdis_test+0x1c: 20 04 18 4e  dup v0.2d, v1.d[1]
    libdis_test+0x20: 00 0c 01 0e  dup v0.8b, w0
    libdis_test+0x24: ff 0f 01 4e  dup v31.16b, wzr
    libdis_test+0x28: 00 0c 02 0e  dup v0.4h, w0
    libdis_test+0x2c: ff 0f 02 4e  dup v31.8h, wzr
    libdis_test+0x30: 00 0c 04 0e  dup v0.2s, w0
    libdis_test+0x34: ff 0f 04 4e  dup v31.4s, wzr
    libdis_test+0x38: 00 0c 08 4e  dup v0.2d, x0
    libdis_test+0x3c: ff 0f 08 4e  dup v31.2d, xzr
    libdis_test+0x40: 00 2c 01 0e  smov w0, v0.b[0]
    libdis_test+0x44: ff 2f 1e 0e  smov wzr, v31.h[7]
    libdis_test+0x48: 00 2c 1f 4e  smov x0, v0.b[15]
    libdis_test+0x4c: ff 2f 1e 4e  smov xzr, v31.h[7]
    libdis_test+0x50: ff 2f 1c 4e  smov xzr, v31.s[3]
    libdis_test+0x54: 00 3c 01 0e  umov w0, v0.b[0]
    libdis_test+0x58: ff 3f 1e 0e  umov wzr, v31.h[7]
    libdis_test+0x5c: 00 3c 08 4e  mov x0, v0.d[0]
    libdis_test+0x60: ff 3f 18 4e  mov xzr, v31.d[1]
    libdis_test+0x64: 00 1c 1f 4e  mov v0.b[15], w0
    libdis_test+0x68: ff 1f 1e 4e  mov v31.h[7], wzr
    libdis_test+0x6c: df 1f 1c 4e  mov v31.s[3], w30
    libdis_test+0x70: ff 1f 18 4e  mov v31.d[1], xzr
    libdis_test+0x74: e0 07 1f 6e  mov v0.b[15], v31.b[0]
    libdis_test+0x78: e0 07 1e 6e  mov v0.h[7], v31.h[0]
    libdis_test+0x7c: e0 07 1c 6e  mov v0.s[3], v31.s[0]
    libdis_test+0x80: e0 07 18 6e  mov v0.d[1], v31.d[0]
    libdis_test+0x84: 00 04 01 5e  mov b0, v0.b[0]
    libdis_test+0x88: ff 07 1f 5e  mov b31, v31.b[15]
    libdis_test+0x8c: 00 04 02 5e  mov h0, v0.h[0]
    libdis_test+0x90: ff 07 1e 5e  mov h31, v31.h[7]
    libdis_test+0x94: 00 04 04 5e  mov s0, v0.s[0]
    libdis_test+0x98: ff 07 1c 5e  mov s31, v31.s[3]
    libdis_test+0x9c: 00 04 08 5e  mov d0, v0.d[0]
    libdis_test+0xa0: ff 07 18 5e  mov d31, v31.d[1]
